### PR TITLE
Merge Jules to Dev

### PR DESCRIPTION
**Pull request by pioupiou-devs**

### CODE REVIEW

File: src/server/models/frame.js [hunk 1]
```diff
@@ -5,6 +5,11 @@ class Frame {
             throw new Error("Invalid data provided");
         }
 
+        //in case of null or undefined input, we set to 0
+        if (c1 == null){c1 = 0;}
```
Comment (pioupiou-devs) [inline src/server/models/frame.js]: Prefer removing braces and return a line

File: src/server/models/grid.js [hunk 2]
```diff
@@ -16,51 +16,77 @@ class Grid {
...
+        //is player existe
+        var frameTable=this.players.get(playerName).frames;
+        if (frameTable == null) {throw new Error('PLayer is not in List');}
+
+        //throw value verification
+        //in case of null or undefined input, we set to 0
+        if (c1 == null){c1 = 0;}
+        if (c2 == null){c2 = 0;}
+        if (c3 == null){c3 = 0;}
+        if (c1 <0 || c1 >this.nbKeel || c2 <0 || c2 >this.nbKeel || c3 <0 || c3 >this.nbKeel){throw new Error('Throw not valid');}
```
Comment (pioupiou-devs) [inline src/server/models/grid.js]: Please explain in a comment

File: src/server/models/grid.js [hunk 3]
```diff
@@ -16,51 +16,77 @@ class Grid {
...
+
+        //frame number verification
+        if (frameNumber <1 || frameNumber >this.nbFrame ){throw new Error('Frame not valid');}
+        this.players.get(playerName).frames[frameNumber -1] = new Frame(c1,c2,c3);
```
Comment (pioupiou-devs) [inline src/server/models/grid.js:59]: Create a variable for `frameNumber-1`

File: test/server/frame.js [hunk 4]
```diff
@@ -168,10 +164,6 @@ describe('Frame', function () {
                 assert.throws(() => new Frame(NaN, NaN, NaN), Error);
             });
 
-            it('should not create a frame with a c1 greater than 10', function () {
-                assert.throws(() => new Frame(11, 0, 0), Error);
```
Comment (pioupiou-devs) [inline test/server/frame.js:172]: You can set the keels to 10 to test it

File: test/server/grid.js [hunk 5]
```diff
@@ -198,30 +199,30 @@ describe('Grid', function () {
             });
         });
 
-        describe('constructFrameList', function () {
+        describe('constructFrameList TODO : Moove to Player tests', function () {
```
Comment (pioupiou-devs) [inline test/server/grid.js]: Move the TODO out of the describe, maybe a comment is enough

Comment (pioupiou-devs) [reply test/server/grid.js]: and please Move not Moove

File: test/server/grid.js [hunk 6]
```diff
@@ -198,30 +199,30 @@ describe('Grid', function () {
...
-                let frameList = grid.constructFrameList();
+                let player = new Player();
+                let frameList = player.constructFrameList();
                 assert.equal(frameList.length, 10);
             });
 
-            it('should return a list of 10 frames with the correct type', function () {
-                let grid = new Grid();
-                let frameList = grid.constructFrameList();
+            it('should return a list of 10 frames with the correct type  TODO : Moove to Player tests', function () {
```
Comment (pioupiou-devs) [inline test/server/grid.js]: again the TODO problem

File: test/server/grid.js [hunk 7]
```diff
@@ -198,30 +199,30 @@ describe('Grid', function () {
...
                 assert.equal(frameList.length, 10);
                 for (let i = 0; i < frameList.length; i++) {
                     assert.equal(frameList[i] instanceof Frame, true);
                 }
             });
 
-            it('should return a list of 10 frames with the correct values', function () {
-                let grid = new Grid();
-                let frameList = grid.constructFrameList();
+            it('should return a list of 10 frames with the correct values  TODO : Moove to Player tests', function () {
```
Comment (pioupiou-devs) [inline test/server/grid.js]: again problem TODO

Comment (zeyzen) [reply test/server/grid.js]: done

File: test/server/grid.js (hunk 6 above)
Comment (zeyzen) [reply test/server/grid.js]: done

File: test/server/grid.js (hunk 5 above)
Comment (zeyzen) [reply test/server/grid.js]: done

File: test/server/frame.js (hunk 4 above)
Comment (zeyzen) [reply test/server/frame.js:172]: will be done later

File: src/server/models/grid.js (hunk 3 above)
Comment (zeyzen) [reply src/server/models/grid.js:59]: as I use only once, it's not reay usefull currently

File: src/server/models/grid.js (hunk 2 above)
Comment (zeyzen) [reply src/server/models/grid.js]: done

File: src/server/models/frame.js (hunk 1 above)
Comment (zeyzen) [reply src/server/models/frame.js]: doen

**Pull request merged**